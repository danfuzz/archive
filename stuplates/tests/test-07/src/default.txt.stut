== true
${ 1 == 1 }
${ 1 == 1.0 }
${ 1 == "1" }
${ 1.0 == 1 }
${ 1.0 == 1.0 }
${ 1.3 == "1.3" }
${ true == true }
${ false == false }
${ true == "true" }
${ false == "false" }
${ null == null }
${ "1" == 1 }
${ "1.5" == 1.5 }
${ "foo" == "foo" }
${ "true" == true }
${ "false" == false }
${ [1] == [1] }
${ [] == [] }

== false
${ 1 == 2 }
${ 1 == 2.0 }
${ 1 == null }
${ 1 == true }
${ 1 == false }
${ 1 == "3" }
${ 12.3 == 13.4 }
${ 12.3 == 12 }
${ 12.3 == null }
${ true == false }
${ false == true }
${ true == 1 }
${ false == 0 }
${ false == null }
${ null == 1 }
${ null == 1.0 }
${ null == 0 }
${ null == "" }
${ null == true }
${ null == false }
${ "1 " == 1 }
${ "1" == [1] }
${ "1" == true }
${ [1] == null }
${ [1] == [2] }
${ [1] == [] }
${ [:] == [] }

!= true
${ 1 != 2 }
${ 1 != 1.1 }
${ 1 != " 1" }
${ 1 != true }
${ 1 != false }
${ 1 != null }
${ 1.0 != 1.1 }
${ 1.0 != 5 }
${ 1.0 != "1.00" }
${ 1.0 != true }
${ 1.0 != false }
${ 1.0 != null }
${ true != false }
${ false != true }
${ true != null }
${ false != null }
${ null != true }
${ null != false }
${ null != 1 }
${ null != 1.1 }
${ null != "null" }
${ null != [] }
${ null != [:] }
${ null != "" }
${ "foo" != "f" }
${ "foo" != "FOO" }
${ "foo" != "" }
${ "foo" != null }
${ "foo" != true }
${ "foo" != false }
${ "" != null }
${ "" != false }
${ "" != true }
${ [] != [:] }
${ [] != null }
${ [:] != [] }
${ [] != null }

!= false
${ 1 != 1 }
${ 1 != 1.0 }
${ 1 != "1" }
${ 2.0 != 2 }
${ 2.0 != 2.0 }
${ 2.3 != "2.3" }
${ false != false }
${ true != true }
${ null != null }
${ "" != "" }
${ "zorch" != "zorch" }
${ "10" != 10 }
${ "1.5" != 1.5 }
${ "true" != true }
${ "false" != false }
${ [] != [] }
${ [1] != [1] }
${ [:] != [:] }

< true
${ 1 < 2 }
${ 1 < 2.5 }
${ 10 < "5" }
${ 2.0 < 2.1 }
${ 2.0 < 3 }
${ 21.3 < "500" }
${ false < "ghoul" }
${ true < "turgid" }
${ "a" < "ab" }
${ "a" < true }
${ "500" < 6 }
${ "" < false }

< false
${ 2 < 2 }
${ 2 < 1 }
${ 2 < 1.5 }
${ 2 < "1" }
${ 2 < null }
${ 2.0 < 2.0 }
${ 2.0 < 1.5 }
${ 2.0 < 1 }
${ 10.1 < "1" }
${ 10.2 < "10.2" }
${ 10.3 < null }
${ true < false }
${ false < true }
${ true < true }
${ false < false }
${ true < null }
${ true < "true" }
${ false < "false" }
${ false < "ass" }
${ true < "butt" }
${ null < null }
${ null < 1 }
${ null < "null" }
${ null < "zap" }
${ "" < "" }
${ "a" < "" }
${ "ab" < "a" }
${ "a" < 1 }
${ "xyzzy" < true }
${ "zap" < false }
${ "boo" < null }

<= true
${ 1 <= 2 }
${ 1 <= 2.5 }
${ 10 <= "5" }
${ 2.0 <= 2.1 }
${ 2.0 <= 3 }
${ 21.3 <= "500" }
${ false <= "ghoul" }
${ true <= "turgid" }
${ "a" <= "ab" }
${ "a" <= true }
${ "" <= false }
${ 1 <= 1 }
${ 1 <= 1.0 }
${ 1 <= "1" }
${ 1.0 <= 1 }
${ 1.0 <= 1.0 }
${ 1.3 <= "1.3" }
${ true <= true }
${ false <= false }
${ true <= "true" }
${ false <= "false" }
${ null <= null }
${ "1" <= 1 }
${ "1.5" <= 1.5 }
${ "foo" <= "foo" }
${ "true" <= true }
${ "false" <= false }
${ [1] <= [1] }
${ [] <= [] }

<= false
${ 2 <= 1 }
${ 2 <= 1.5 }
${ 2 <= "1" }
${ 2 <= null }
${ 2 <= true }
${ 2.0 <= 1.5 }
${ 2.0 <= 1 }
${ 10.1 <= "1" }
${ 10.2 <= false }
${ false <= "ass" }
${ true <= "butt" }
${ false <= null }
${ "a" <= "" }
${ "ab" <= "a" }
${ "5" <= 4 }
${ "xyzzy" <= true }
${ "zap" <= false }
${ "x" <= null }

> true
${ 2 > 1 }
${ 2 > 1.5 }
${ "5" > 10 }
${ 2.1 > 2.0 }
${ 3 > 2.0 }
${ "500" > 21.3 }
${ "ghoul" > false }
${ "turgid" > true }
${ "ab" > "a" }
${ true > "a" }
${ false > "" }

> false
${ 2 > 2 }
${ 1 > 2 }
${ 1.5 > 2 }
${ "1" > 2 }
${ null > 2 }
${ 2.0 > 2.0 }
${ 1.5 > 2.0 }
${ 1 > 2.0 }
${ "1" > 10.1 }
${ "10.2" > 10.2 }
${ null > 10.3 }
${ false > true }
${ true > false }
${ true > true }
${ false > false }
${ null > true }
${ "true" > true }
${ "false" > false }
${ "ass" > false }
${ "butt" > true }
${ null > null }
${ 1 > null }
${ "null" > null }
${ "zap" > null }
${ "" > "" }
${ "" > "a" }
${ "a" > "ab" }
${ 1 > "a" }
${ true > "xyzzy" }
${ false > "zap" }
${ null > "boo" }

>= true
${ 2 >= 1 }
${ 2.5 >= 1 }
${ "5" >= 10 }
${ 2.1 >= 2.0 }
${ 3 >= 2.0 }
${ "500" >= 21.3 }
${ "ghoul" >= false }
${ "turgid" >= true }
${ "ab" >= "a" }
${ true >= "a" }
${ false >= "" }
${ 1 >= 1 }
${ 1.0 >= 1 }
${ "1" >= 1 }
${ 1 >= 1.0 }
${ 1.0 >= 1.0 }
${ "1.3" >= 1.3 }
${ true >= true }
${ false >= false }
${ "true" >= true }
${ "false" >= false }
${ null >= null }
${ 1 >= "1" }
${ 1.5 >= "1.5" }
${ "foo" >= "foo" }
${ true >= "true" }
${ false >= "false" }
${ [1] >= [1] }
${ [] >= [] }

>= false
${ 1 >= 2 }
${ 1.5 >= 2 }
${ "1" >= 2 }
${ null >= 2 }
${ true >= 2 }
${ 1.5 >= 2.0 }
${ 1 >= 2.0 }
${ "1" >= 10.1 }
${ false >= 10.2 }
${ "ass" >= false }
${ "butt" >= true }
${ null >= false }
${ "" >= "a" }
${ "a" >= "ab" }
${ 4 >= "5" }
${ true >= "xyzzy" }
${ false >= "zap" }
${ null >= "x" }

!
${ !true } false
${ !false } true

&
${ 0xff & 0xdb } 219
${ 0x00 & 0xdb } 0
${ 0x00 & 0x00 } 0
${ false & false } false
${ false & true } false
${ true & false } false
${ true & true } true

|
${ 0xff | 0xdb } 255
${ 0x00 | 0xdb } 219
${ 0x00 | 0x00 } 0
${ false | false } false
${ false | true } true
${ true | false } true
${ true | true } true

^
${ 0xff ^ 0xdb } 36
${ 0x00 ^ 0xdb } 219
${ 0x00 ^ 0x00 } 0
${ false ^ false } false
${ false ^ true } true
${ true ^ false } true
${ true ^ true } false

~
${ ~true } false
${ ~false } true
${ ~0 } -1
${ ~219 } -220

<<
${ 0 << 10 } 0
${ 1 << 1 } 2
${ 10 << 2 } 40
${ 10 << -1 } 5
${ 123 << 0 } 123
${ -1 << 1 } -2
${ -15 << 2 } -60

<<
${ 0 >> 10 } 0
${ 1 >> 1 } 0
${ 10 >> 2 } 2
${ 10 >> -1 } 20
${ 123 >> 0 } 123
${ -1 >> 1 } -1
${ -60 >> 2 } -15

+
${ 0 + 0 } 0
${ 0 + 2 } 2
${ 2 + 0 } 2
${ 2 + 2 } 4
${ 0.0 + 0.0 } 0.0
${ 0.0 + 1.0 } 1.0
${ 2.0 + 0.0 } 2.0
${ 3.0 + 4.0 } 7.0
${ 1 + 1.5 } 2.5
${ 1.5 + 1 } 2.5
expect nothing: ${ "" + "" }
${ "a" + "b" } ab
${ "a" + 1 } a1
${ 1.2 + "a" } 1.2a
${ "a" + null } anull
${ null + "a" } nulla
${ "a" + true } atrue
${ false + "a" } falsea
${ null + null } nullnull
${ 4 + null } 4null
${ null + 3.2 } null3.2
expect nothing: ${ +null }
${ +5 } 5
${ +"foo" } foo

-
${ 0 - 0 } 0
${ 0 - 1 } -1
${ 3 - 1.0 } 2.0
${ 0.0 - 0.0 } 0.0
${ 3.0 - 1.5 } 1.5
${ 2.0 - 1 } 1.0
${ -0 } 0
${ -1 } -1
${ -0.0 } -0.0
${ -1.0 } -1.0

*
${ 0 * 0 } 0
${ 2 * 3 } 6
${ 2 * 1.0 } 2.0
${ 0.0 * 0.0 } 0.0
${ 5.1 * 2.0 } 10.2
${ 1.2 * 2 } 2.4

/
${ 0 / 1 } 0
${ 1 / 2 } 0
${ 4 / 2 } 2
${ 1 / 2.0 } 0.5
${ 0.0 / 1.0 } 0.0
${ 1.1 / 2.0 } 0.55
${ 5.0 / 2 } 2.5

%
${ 5 % 2 } 1
${ 5 % 2.0 } 1.0
${ 5.0 % 2.0 } 1.0
${ 5.0 % 2 } 1.0
${ -5 % 2 } -1
${ -5 % 2.0 } -1.0
${ -5.0 % 2.0 } -1.0
${ -5.0 % 2 } -1.0
${ 5 % -2 } 1
${ 5 % -2.0 } 1.0
${ 5.0 % -2.0 } 1.0
${ 5.0 % -2 } 1.0
${ -5 % -2 } -1
${ -5 % -2.0 } -1.0
${ -5.0 % -2.0 } -1.0
${ -5.0 % -2 } -1.0

%%
${ 5 %% 2 } 1
${ 5 %% 2.0 } 1.0
${ 5.0 %% 2.0 } 1.0
${ 5.0 %% 2 } 1.0
${ -5 %% 2 } 1
${ -5 %% 2.0 } 1.0
${ -5.0 %% 2.0 } 1.0
${ -5.0 %% 2 } 1.0

**
${ 2 ** 2 } 4
${ 2 ** 2.0 } 4.0
${ 2.0 ** 2.0 } 4.0
${ 2.0 ** 2 } 4.0

[]
${ [1,2][0] } 1
${ ["a":"b", "c":"d"]["c"] } d
