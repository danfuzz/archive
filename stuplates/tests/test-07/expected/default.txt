== true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true

== false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false

!= true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true

!= false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false

< true
true
true
true
true
true
true
true
true
true
true
true
true

< false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false

<= true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true

<= false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false

> true
true
true
true
true
true
true
true
true
true
true
true

> false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false

>= true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true

>= false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false
false

!
false false
true true

&
219 219
0 0
0 0
false false
false false
false false
true true

|
255 255
219 219
0 0
false false
true true
true true
true true

^
36 36
219 219
0 0
false false
true true
true true
false false

~
false false
true true
-1 -1
-220 -220

<<
0 0
2 2
40 40
5 5
123 123
-2 -2
-60 -60

<<
0 0
0 0
2 2
20 20
123 123
-1 -1
-15 -15

+
0 0
2 2
2 2
4 4
0.0 0.0
1.0 1.0
2.0 2.0
7.0 7.0
2.5 2.5
2.5 2.5
expect nothing: 
ab ab
a1 a1
1.2a 1.2a
anull anull
nulla nulla
atrue atrue
falsea falsea
nullnull nullnull
4null 4null
null3.2 null3.2
expect nothing: 
5 5
foo foo

-
0 0
-1 -1
2.0 2.0
0.0 0.0
1.5 1.5
1.0 1.0
0 0
-1 -1
-0.0 -0.0
-1.0 -1.0

*
0 0
6 6
2.0 2.0
0.0 0.0
10.2 10.2
2.4 2.4

/
0 0
0 0
2 2
0.5 0.5
0.0 0.0
0.55 0.55
2.5 2.5

%
1 1
1.0 1.0
1.0 1.0
1.0 1.0
-1 -1
-1.0 -1.0
-1.0 -1.0
-1.0 -1.0
1 1
1.0 1.0
1.0 1.0
1.0 1.0
-1 -1
-1.0 -1.0
-1.0 -1.0
-1.0 -1.0

%%
1 1
1.0 1.0
1.0 1.0
1.0 1.0
1 1
1.0 1.0
1.0 1.0
1.0 1.0

**
4 4
4.0 4.0
4.0 4.0
4.0 4.0

[]
1 1
d d
